src/serac/numerics/functional/quadrature_data.hpp

```diff
@@ -54,54 +107,64 @@ struct Empty {
  */
 template <typename T>
 struct QuadratureData {
-  /// ctor, allocates memory and sets up strides
-  QuadratureData(size_t n1, size_t n2) : stride(n2) { data = new T[n1 * n2]; }
-
-  /// dtor, deallocates memory
-  ~QuadratureData() { delete data; }
 
-  /// access a mutable reference to the quadrature data at element `i`, quadrature point `j`
-  SERAC_HOST_DEVICE T& operator()(size_t i, size_t j) { return data[i * stride + j]; }
+  using geom_array_t = std::array< uint32_t, mfem::Geometry::NUM_GEOMETRIES >;
 
-  /// access a const reference to the quadrature data at element `i`, quadrature point `j`
-  SERAC_HOST_DEVICE const T& operator()(size_t i, size_t j) const { return data[i * stride + j]; }
+  using tmp_t = axom::Array<T, 2>;
 
-  T*     data;    ///< pointer to the buffer of quadrature data
-  size_t stride;  ///< how many quadrature points per element
+  /// ctor, allocates memory and sets up strides
+  QuadratureData(geom_array_t elements, geom_array_t qpts_per_element, T value = T{}) { 
+
+    constexpr std::array geometries = {
+      mfem::Geometry::SEGMENT,
+      mfem::Geometry::TRIANGLE,
+      mfem::Geometry::SQUARE,
+      mfem::Geometry::TETRAHEDRON,
+      mfem::Geometry::CUBE
+    };
+
+    for (auto geom : geometries) {
+      if (elements[uint32_t(geom)] > 0) {
+        data[geom] = tmp_t(elements[uint32_t(geom)], qpts_per_element[uint32_t(geom)]);
+        data[geom].fill(value);
+      }
+    }
+
+  }
+
+  axom::ArrayView<T, 2> operator[](mfem::Geometry::Type geom) {
+    return axom::ArrayView<T, 2>(data.at(geom));
+  }
+
+  std::map< mfem::Geometry::Type, tmp_t > data;
```

Comment:
What are the indicies of this `data` object? I assume [geometry_type][element_number][quadrature_index].